### PR TITLE
Adding features to calendar

### PR DESCRIPTION
**Pull request by macintosh1398**
This PR adds new features to the calendar (and also correct bugs on date selection and database connection). The user is now able to add directly a deadline using only the keyboard (and press ENTER), which is way more convenient. Another feature that turns the app more intuitive is the ability for the user to escape the text input by just touching outside it. The calendar now communicates with Firebase in a different manner (team refactoring).

### CODE REVIEW

**Comment by macintosh1398** [comment]
Displaying deadlines on the calendar view is way more difficult than I thought. After many hours of research, it will not be done this week.

File: app/src/androidTest/java/com/github/multimatum_team/multimatum/CalendarActivityTest.kt [hunk 1]
```diff
@@ -38,6 +45,30 @@ class CalendarActivityTest {
...
+            .perform(ViewActions.closeSoftKeyboard())
+        Espresso.onView(ViewMatchers.withId(R.id.calendar_add_deadline_button))
+            .perform(click())
+        Espresso.onView(ViewMatchers.withId(R.id.textInputEditCalendar))
+            .check(ViewAssertions.matches(ViewMatchers.isClickable()))
+    }
+
+    /*
+    @Test
+    fun keyboardDisappearWhenTouchedOutside() {
```
Comment (Matathiasq13824) [inline app/src/androidTest/java/com/github/multimatum_team/multimatum/CalendarActivityTest.kt:62]: Why this test is in a comment ?

File: app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt [hunk 2]
```diff
@@ -1,54 +1,114 @@
...
 
         // Getting the entered text and the selected date
         val deadlineTitle = editText.text.toString()
-        val selectedDate =
-            Instant.ofEpochMilli(calendar.date).atZone(ZoneId.systemDefault()).toLocalDate()
 
         // Add the new event to the deadline list
         val deadline = Deadline(deadlineTitle, DeadlineState.TODO, selectedDate)
+
         viewModel.addDeadline(deadline)
```
Comment (Matathiasq13824) [inline app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt:112]: I see that you use now the viewModel instead of the repository. Why this change ?

File: app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt [hunk 3]
```diff
@@ -1,54 +1,114 @@
...
+    an eventual date selection by the user.
+     */
     private fun initCalendar() {
         setContentView(R.layout.activity_calendar)
+
+        // Get the calendar view
+        val calendar = findViewById<CalendarView>(R.id.calendar_view)
+
+        // Update the selected date in case of a selection in the calendar
+        calendar.setOnDateChangeListener { _, year, month, day ->
```
Comment (Matathiasq13824) [inline app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt:48]: It is possible to test this to increase coverage ?

File: app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt [hunk 4]
```diff
@@ -1,54 +1,114 @@
...
 import com.google.android.material.textfield.TextInputEditText
 import dagger.hilt.android.AndroidEntryPoint
-import kotlinx.coroutines.runBlocking
 import java.time.Instant
+import java.time.LocalDate
 import java.time.ZoneId
-import javax.inject.Inject
+
+
+//import javax.inject.Inject
```
Comment (Matathiasq13824) [inline app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt:24]: If this is not useful, you can delete this comment or make it closer to the other import I think.

**Review by Matathiasq13824 — COMMENTED** [review]
I have some question about some of the modifications done, however it's a grate job and a nice code

**Review by LennyDZ — APPROVED** [review]

File: app/src/androidTest/java/com/github/multimatum_team/multimatum/CalendarActivityTest.kt (hunk 1 above)
Comment (macintosh1398) [reply app/src/androidTest/java/com/github/multimatum_team/multimatum/CalendarActivityTest.kt:72]: It was a tentative for another UI test, I removed it because it is far more complex than I thought. It will be done in the coming weeks.

File: app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt (hunk 3 above)
Comment (macintosh1398) [reply app/src/main/java/com/github/multimatum_team/multimatum/CalendarActivity.kt:48]: Good point. But honestly, I don't know how to test it. The CalendarView is a very complex structure (containing a lot of sub-structures). To test that, we need to be able to simulate a date selection on the calendar, and I don't know how to do it. If someone know how to do that, feel free to tell me.

**Review by Matathiasq13824 — APPROVED** [review]
